File: Server/SocialLogin/Models/Login/SignInContent.cs
using System.Collections.Generic;


namespace SocialLogin.Models.Login
{
    public class SignInContent
    {
        //public string AuthToken { get; set; }
        public List<string> SignInOption { get; set; }
        public string MobileConfigUrl { get; set; }
        public int CustomerListId { get; set; }
        public SignIn SignIn { get; set; }
        public Footer Footer { get; set; }
        public Images Images { get; set; }
        public Icons Icons { get; set; }
    }
}

File: Client/src/models/initial/signIn/signInContent.ts
import { Record, List } from 'immutable';
import { convertListToImmutable } from 'utils/immutable';
import { SignIn, Footer, Images, Icons } from './';

interface Interface {
  signInOption: List<string>,
  mobileConfigUrl: string,
  customerListId: number,
  signIn?: SignIn,
  footer?: Footer,
  images?: Images,
  icons?: Icons
}

const initialValue = Record<Interface>({
  signInOption: List<string>(),
  mobileConfigUrl: '',
  customerListId: 0,
  signIn: undefined,
  footer: undefined,
  images: undefined,
  icons: undefined
});

export class SignInContent extends initialValue {
  constructor(args: any = {}) {
    super(Object.assign({}, args as Interface, {
      signIn: args.signIn && new SignIn(args.signIn),
      footer: args.footer && new Footer(args.footer),
      images: args.images && new Images(args.images),
      icons: args.icons && new Icons(args.icons)
    }));
  }
}
